week2/JS/sig_all/utils/sig_all_func.js

```diff
@@ -0,0 +1,70 @@
+import { email, password, checkPassword, passwordInputs, testData } from "../contants.js";
+
+const errText = document.createElement('p');
+errText.classList.add('errText')
+
+function isValidEmail(email) {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+};
+
+function isValidPassword(password) {
+  const passwordRegex = /^(?=.*\d)(?=.*[a-zA-Z])[A-Za-z\d]{8,}$/;
+  return passwordRegex.test(password);
+}
+
+
+function emailError (e) {
+  if (e.target === email) {
+    if (e.target.value === '') {
+      errText.textContent = '이메일을 입력해 주세요.';
+    } else if (!isValidEmail(e.target.value)) {
+      errText.textContent = '올바른 이메일 주소가 아닙니다.';
+    } else if(testData.some((member) => member.email === email.value) && checkPassword) {
+      errText.textContent = '이미 사용 중인 이메일입니다.';
+    }  
+    e.target.classList.add('err')
+    e.target.after(errText);
+  }
+}
```

Comment:
`early return`을 사용해 코드의 depth를 줄여 가독성을 높여보시면 좋을거 같네요~

```suggestion
function emailError (e) {
  if(e.target !== email) return;


    if (e.target.value === '') {
      errText.textContent = '이메일을 입력해 주세요.';
    } else if (!isValidEmail(e.target.value)) {
      errText.textContent = '올바른 이메일 주소가 아닙니다.';
    } else if(testData.some((member) => member.email === email.value) && checkPassword) {
      errText.textContent = '이미 사용 중인 이메일입니다.';
    }  
    e.target.classList.add('err')
    e.target.after(errText);
}
```